app/controllers/course/experience_points_records_controller.rb

```diff
@@ -1,21 +1,42 @@
 # frozen_string_literal: true
 class Course::ExperiencePointsRecordsController < Course::ComponentController
-  load_resource :course_user, through: :course, id_param: :user_id
+  load_resource :course_user, through: :course, id_param: :user_id, except: [:read_all_exp, :download]
   load_and_authorize_resource :experience_points_record, through: :course_user,
-                                                         class: Course::ExperiencePointsRecord.name
+                                                         class: Course::ExperiencePointsRecord.name,
+                                                         except: [:read_all_exp, :download]
+
+  def read_all_exp
+    if filter_and_page_params[:student_id].present?
```

Comment:
To make it more succinct, why not
`course_user_id = filter_and_page_params[:student_id].present? ? filter_and_page_params[:student_id] : @course.course_users.pluck(:id)`
`@experience_points_records = Course::ExperiencePointsRecord.where(course_user_id: course_user_id)`

Reply:
And why duplicate code with `fetch_experience_points_records`?